File: NetcraftServer/Commands/Commandgive.vb
Imports NetcraftServer

Public Class Commandgive
    Inherits Command
    Sub New()
        MyBase.New("give", "Выдать предмет игроку/себе", "give <игрок|@s|@a|@r> <предмет> [кол-во]")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.GetAdmin Then
            sender.SendMessage("У Вас недостаточно прав!")
            Return True
        End If
        If args.Length = 2 Then
            Dim t As Material = [Enum].Parse(GetType(Material), args(1).ToUpper)
            Dim p As NetworkPlayer
            If args(0) = "@s" Then
                If Not sender.IsPlayer Then
                    sender.SendMessage("Только для игрока!")
                    Return True
                End If
                p = sender
                p.Give(t)
                p.Chat($"Выдано {t.ToString.ToLower} (1 шт.) игроку " + p.Username)
                Return True
            ElseIf args(0) = "@a" Then
                For Each g In netcraft.server.api.NCSApi.Netcraft.GetOnlinePlayers
                    g.Give(t)
                Next
                sender.SendMessage($"Выдано {t.ToString.ToLower} (1 шт.) {netcraft.server.api.NCSApi.Netcraft.GetOnlinePlayers.Count} игрокам")
                Return True
            ElseIf args(0) = "@r" Then
                Dim rnd As Random = New Random
                Dim g = netcraft.server.api.NCSApi.Netcraft.GetOnlinePlayers()
                p = g(rnd.Next(0, g.Count - 1))
                p.Give(t)
                sender.SendMessage($"Выдано {t.ToString.ToLower} (1 шт.) игроку " + p.Username)
                Return True
            Else

                p = netcraft.server.api.NCSApi.Netcraft.GetPlayer(args(0))
                If IsNothing(p) Then
                    sender.SendMessage("Игрок не найден!")
                    Return True
                End If
                p.Give(t)
                sender.SendMessage($"Выдано {t.ToString.ToLower} (1 шт.) игроку " + p.Username)
                Return True
            End If

        End If


        If args.Length = 3 Then
            Dim t As Material = [Enum].Parse(GetType(Material), args(1).ToUpper)
            Dim count As Integer = CInt(args(2))
            Dim p As NetworkPlayer
            If args(0) = "@s" Then
                If Not sender.IsPlayer Then
                    sender.SendMessage("Только для игрока!")
                    Return True
                End If
                p = sender
                p.Give(t, count)
                p.Chat($"Выдано {t.ToString.ToLower} ({count} шт.) игроку " + p.Username)
                Return True
            ElseIf args(0) = "@a" Then
                For Each g In netcraft.server.api.NCSApi.Netcraft.GetOnlinePlayers
                    g.Give(t, count)
                Next
                sender.SendMessage($"Выдано {t.ToString.ToLower} ({count} шт.) {netcraft.server.api.NCSApi.Netcraft.GetOnlinePlayers.Count} игрокам")
                Return True
            ElseIf args(0) = "@r" Then
                Dim rnd As Random = New Random
                Dim g = netcraft.server.api.NCSApi.Netcraft.GetOnlinePlayers()
                p = g(rnd.Next(0, g.Count - 1))
                p.Give(t, count)
                sender.SendMessage($"Выдано {t.ToString.ToLower} ({count} шт.) игроку " + p.Username)
                Return True
            Else

                p = netcraft.server.api.NCSApi.Netcraft.GetPlayer(args(0))
                If IsNothing(p) Then
                    sender.SendMessage("Игрок не найден!")
                    Return True
                End If
                p.Give(t, count)
                sender.SendMessage($"Выдано {t.ToString.ToLower} ({count} шт.) игроку " + p.Username)
                Return True
            End If

        End If
        Return False
    End Function
End Class

File: NCore/Commands/Commandgive.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.VisualBasic.CompilerServices;
using NCore.netcraft.server.api;

namespace NCore
{
    public class Commandgive : Command
    {
        public Commandgive() : base("give", NCore.GetNCore().lang.get("commands.give.description"), NCore.GetNCore().lang.get("commands.give.usage"))
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            NCore.Lang lang = sender.IsPlayer ? ((NetcraftPlayer)sender).lang : NCore.GetNCore().lang;
            if (!sender.GetAdmin())
            {
                await sender.SendMessage(lang.get("commands.generic.error.no-perms"));
                return true;
            }

            if (args.Length == 2)
            {
                args[2] = "1";
            }

            if (args.Length == 3)
            {
                Material t = (Material)Enum.Parse(typeof(Material), args[1].ToUpper());
                int count = Conversions.ToInteger(args[2]);
                NetcraftPlayer p;
                if (args[0] == "@s")
                {
                    if (!sender.IsPlayer)
                    {
                        await sender.SendMessage("Только для игрока!");
                        return true;
                    }

                    p = (NetcraftPlayer)sender;
                    await p.Give(t, count);
                    await NCore.GetNCore().SendCommandFeedback(lang.get("commands.give.success.player", t.ToString().ToLower(), args[2], p.Username), sender);
                    return true;
                }
                else if (args[0] == "@a")
                {
                    foreach (var g in Netcraft.GetOnlinePlayers())
                        await g.Give(t, count);
                    await NCore.GetNCore().SendCommandFeedback(lang.get("commands.give.success.all", t.ToString().ToLower(), args[2], Netcraft.GetOnlinePlayers().Count.ToString()), sender);
                    return true;
                }
                else if (args[0] == "@r")
                {
                    var rnd = new Random();
                    var g = Netcraft.GetOnlinePlayers();
                    p = g[rnd.Next(0, g.Count - 1)];
                    await p.Give(t, count);
                    await NCore.GetNCore().SendCommandFeedback(lang.get("commands.give.success.player", t.ToString().ToLower(), args[2], p.Username), sender);
                    return true;
                }
                else
                {
                    p = Netcraft.GetPlayer(args[0]);
                    if (NCore.IsNothing(p))
                    {
                        await sender.SendMessage(lang.get("commands.generic.player.not-found"));
                        return true;
                    }

                    await p.Give(t, count);
                    await NCore.GetNCore().SendCommandFeedback(lang.get("commands.give.success.player", t.ToString().ToLower(), args[2], p.Username), sender);
                    return true;
                }
            }

            return false;
        }
    }
}